packages/run-protocol/src/psm/psm.js

```diff
@@ -0,0 +1,170 @@
+
+import { AmountMath } from '@agoric/ertp';
+import { makeMakeCollectFeesInvitation } from '../vaultFactory/collectRewardFees.js';
+
+const { details: X } = assert;
+
+const BASIS_POINTS = 10000n;
+
+/**
+ * Stage a transfer of a singe asset from one seat to another, with an optional
+ * remapping of the Keywords. Check that the remapping is for the same amount.
+ *
+ * @param {ZCFSeat} from
+ * @param {ZCFSeat} to
+ * @param {AmountKeywordRecord} txFrom
+ * @param {AmountKeywordRecord} txTo
+ */
+function stageTransfer(from, to, txFrom, txTo = txFrom) {
+  assert(AmountMath.isEqual(Object.values(txFrom)[0], Object.values(txTo)[0]));
+  from.decrementBy(txFrom);
+  to.incrementBy(txTo);
+}
+
+/**
+ * @param {ZCF<{
+ *    anchorBrand: Brand,
+ *    main: {
+ *      WantStableFeeBP: bigint,
+ *      GiveStableFeeBP: bigint,
+ *      MintLimit: Amount } }>
+ * } zcf
+ * @param {{feeMintAccess: FeeMintAccess}} privateArgs
+ */
+export const start = async (zcf, privateArgs) => {
+  const { anchorBrand } = zcf.getTerms();
+
+  const { feeMintAccess } = privateArgs;
+  // TODO should this know that the name is 'Stable'
+  // TODO get the RUN magic out of here so the contract is more reusable
+  const stableMint = await zcf.registerFeeMint('Stable', feeMintAccess);
+  const { brand: stableBrand } = stableMint.getIssuerRecord();
+  zcf.setTestJig(() => ({
+    stableIssuerRecord: stableMint.getIssuerRecord(),
+  }));
+  const emptyStable = AmountMath.makeEmpty(stableBrand);
+  const emptyAnchor = AmountMath.makeEmpty(anchorBrand);
+
+  // HACK for simple governance API
+  const getGovernance = _ => {
+    const {
+      main: { WantStableFeeBP, GiveStableFeeBP, MintLimit },
+    } = zcf.getTerms();
+    assert(
+      AmountMath.isGTE(MintLimit, emptyAnchor),
+      X`MintLimit ${MintLimit} must specificy a limit in ${anchorBrand}`,
+    );
+    return {
+      getWantStableRate: () =>
+        makeRatio(WantStableFeeBP, stableBrand, BASIS_POINTS),
+      getGiveStableRate: () =>
+        makeRatio(GiveStableFeeBP, stableBrand, BASIS_POINTS),
+      getMintLimit: () => MintLimit,
+    };
+  };
+  const gov = getGovernance(zcf);
+
+  const { zcfSeat: anchorPool } = zcf.makeEmptySeatKit();
+  const { zcfSeat: feePool } = zcf.makeEmptySeatKit();
+  const { zcfSeat: stage } = zcf.makeEmptySeatKit();
+
+  const assertUnderLimit = anchorAmount => {
+    assert(
+      AmountMath.isGTE(gov.getMintLimit(), anchorAmount),
+      X`Request would exceed mint limit`,
+    );
+  };
+
+  const giveStable = (seat, given, wanted = emptyAnchor) => {
+    const fee = ceilMultiplyBy(given, gov.getGiveStableRate());
+    const afterFee = AmountMath.subtract(given, fee);
+    const maxAnchor = AmountMath.make(anchorBrand, afterFee.value);
+    // TODO this prevents the reallocate from failing. Can this be tested otherwise?
+    assert(
+      AmountMath.isGTE(maxAnchor, wanted),
+      X`wanted ${wanted} is more then ${given} minus fees ${fee}`,
+    );
```

Comment:
This is just because it was paralleling the other branch, which used the assert to avoid a spurious minting event.

Reply:
Either way, WDYT of a `@throws` comment on the function jsdoc to describe?

Reply:
I shall avoid @throws both because of Java PTSD (pervasive throw system disorders) and because we are still debating exactly how to abruptly terminate.

Reply:
What's the status of resolving that debate?

When it's resolved, how will we know to come back and make this operate that way?

Reply:
There's a separate discussion about 1. using `assert` vs. 2. use `assert` here instead of relying on reallocate. 
WRT 1: Vaults use asserts, so any change to that pattern will go through all of RUN protocol and change it.
WRT 2: I think that the details that includes fees will be informative in the easy mistake case that someone says "give 2 USDC want 2 RUN"

So I concluded that I should leave it.